File: src/main/java/no/hvl/tk/visual/debugger/server/UIServerStarter.java
package no.hvl.tk.visual.debugger.server;

import com.intellij.openapi.diagnostic.Logger;
import org.glassfish.grizzly.http.server.CLStaticHttpHandler;
import org.glassfish.grizzly.http.server.HttpServer;
import org.glassfish.grizzly.http.server.NetworkListener;

public class UIServerStarter {
    private static final Logger LOGGER = Logger.getInstance(UIServerStarter.class);

    private UIServerStarter() {
        // Only helper methods.
    }

    public static HttpServer runNewServer() {
        final HttpServer server = new HttpServer();
        final NetworkListener networkListener = new NetworkListener(
                "UI",
                ServerConstants.HOST_NAME,
                ServerConstants.UI_SERVER_PORT);
        server.addListener(networkListener);

        server.getServerConfiguration().addHttpHandler(
                new CLStaticHttpHandler(
                        UIServerStarter.class.getClassLoader(),
                        ServerConstants.STATIC_RESOURCE_PATH),
                "/");
        try {
            server.start();
            LOGGER.info("UI server started successfully.");
            return server;
        } catch (final Exception e) {
            LOGGER.error(e);
            return null;
        }
    }
}

File: src/main/java/no/hvl/tk/visual/debugger/server/UIServerStarter.kt
package no.hvl.tk.visual.debugger.server

import com.intellij.openapi.diagnostic.Logger
import java.io.IOException
import org.glassfish.grizzly.http.server.CLStaticHttpHandler
import org.glassfish.grizzly.http.server.HttpServer
import org.glassfish.grizzly.http.server.NetworkListener

object UIServerStarter {
  private val LOGGER = Logger.getInstance(UIServerStarter::class.java)

  @JvmStatic
  fun runNewServer(): HttpServer? {
    val server = HttpServer()
    val networkListener =
        NetworkListener("UI", ServerConstants.HOST_NAME, ServerConstants.UI_SERVER_PORT)
    server.addListener(networkListener)

    server.serverConfiguration.addHttpHandler(
        CLStaticHttpHandler(
            UIServerStarter::class.java.classLoader, ServerConstants.STATIC_RESOURCE_PATH),
        "/")
    try {
      server.start()
      LOGGER.info("UI server started successfully.")
      return server
    } catch (e: IOException) {
      LOGGER.error(e)
      return null
    }
  }
}
